code/Control/UserDefinedFormController.php

```diff
@@ -355,7 +355,8 @@ public function process($data, $form)
 
                         // attach a file to recipient email only if lower than configured size
                         if ($file->getAbsoluteSize() <= $this->getMaximumAllowedEmailAttachmentSize()) {
-                            $attachments[] = $file;
+                            // using the field name as array index is fine as file upload field only allows one file
+                            $attachments[$field->Name] = $file;
```

Comment:
If this changes to allow multiple files in the future, it will be easy enough to change this to `$attachments[$field->Name][] = $file;`, so no worries here.